tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/QueryFixerMain.java

```diff
@@ -1,227 +1,75 @@
-      System.out.println("The input query is valid. No errors to fix.");
-      return;
-    }
-
-    System.out.println("The query has an error: " + fixResult.getError());
 
-    if (fixResult.getStatus() == FixResult.Status.FAILURE) {
-      System.out.println("Failed to fix the input query.");
-      return;
-    }
-
-    System.out.println("It can be fixed by the approach: " + fixResult.getApproach());
-    printFixOptions(fixResult.getOptions());
-  }
-
-  private static void printFixOptions(List<FixOption> options) {
-    System.out.println();
-    int count = 1;
-    for (FixOption option : options) {
-      System.out.println(String.format("%d. Option: %s", count++, option.getDescription()));
-      System.out.println(String.format("   Fixed query: %s", option.getFixedQuery()));
-      System.out.println();
+    // If no credentials is provided, the program uses:
+    // <ol>
+    // <li> the default path in the Env variable: GOOGLE_APPLICATION_CREDENTIALS.
+    // <li> Default credentials location (OS dependent). For example,
+    // "~/.config/gcloud/application_default_credentials.json" in Linux. Usually, it should be same
+    // path of the credentials you create through calling the command "gcloud auth
+    // application-default login".
+    // </ol>
+    if (credentialsPath == null) {
+      return BigQueryOptions.newBuilder().setProjectId(projectId).build();
     }
-  }
 
-  private static void printQueryResult(String query, BigQueryOptions options) {
-    BigQuery bigQuery = options.getService();
-
-    QueryJobConfiguration queryConfig = QueryJobConfiguration.newBuilder(query).build();
-    try {
-      int maxCount = 10;
-      int count = 0;
-      System.out.println("=".repeat(20));
-      System.out.println("Now query: " + query);
-      System.out.printf("Only %d rows are printed\n", maxCount);
-      TableResult result = bigQuery.query(queryConfig);
-
-      for (Field field : result.getSchema().getFields()) {
-        System.out.printf("%s,", field.getName());
-      }
-      System.out.print("\n");
-
-      for (FieldValueList row : bigQuery.query(queryConfig).iterateAll()) {
-        for (FieldValue val : row) {
-          System.out.printf("%s,", val.getValue());
-        }
-        System.out.print("\n");
-        if (++count >= maxCount) {
-          break;
-        }
-      }
-    } catch (InterruptedException ignored) {
+    File credentialsFile = new File(credentialsPath);
+    GoogleCredentials credentials;
+    try (FileInputStream serviceAccountStream = new FileInputStream(credentialsFile)) {
+      credentials = ServiceAccountCredentials.fromStream(serviceAccountStream);
+    } catch (IOException e) {
+      System.out.println(
+          "Cannot read the credentials from the flag -c. Please provide a correct credentials path with read permission.");
+      exit(1);
+      // The program will never go down, but the compiler needs a extra RETURN to realize it.
```

Comment:
nit: will never reach to the point, return null is enforced by compilation.

Reply:
Done